binary_search_tree/tree.py

```diff
@@ -14,44 +14,157 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
-    def add(self, key, value = None):
-        pass
+    def add_helper(self, current_node, key, value):
+        if current_node == None:
+            return TreeNode(key, value)
 
-    # Time Complexity: 
-    # Space Complexity: 
+        if key <= current_node.key:
+            current_node.left = self.add_helper(current_node.left, key, value)
+        else:
+            current_node.right = self.add_helper(current_node.right, key, value)
+        return current_node
+        
+
+
+    # Time Complexity: O(log n)
+    # Space Complexity: O(1)
+    def add(self, key, value=None):
```

Comment:
👍 However since the `add_helper` is recursive, this function is O(log n) for space complexity.